### ISSUE TITLE
No reports on vue-nativescript (NS 8)

### ISSUE THREAD

**Issue by VepretskiyD** [nativescript-community/sentry#4]
### Which platform(s) does your issue occur on?
- Android
- 7 (emulator), 11 (device) 
- both emulator and device

### Please, provide the following version numbers that your issue occurs with:

- CLI: 8.0.2
- Cross-platform modules: 8.0.8
- Runtime(s): Android 8.0.0
- Plugin(s): 
```  
  "dependencies": {
    "@nativescript-community/sentry": "^2.1.0",
    "@nativescript/core": "~8.0.0",
    "@nativescript/theme": "~3.0.1",
    "nativescript-vue": "~2.9.0"
  },
  "devDependencies": {
    "@nativescript/android": "8.0.0",
    "@nativescript/webpack": "beta",
    "nativescript-vue-template-compiler": "~2.9.0",
    "sass": "^1.32.8"
  },
```

### Please, tell us how to recreate the issue in as much detail as possible. 
```
ns create my-blank-vue --template @nativescript/template-blank-vue
cd my-blank-vue
npm i
tns plugin add @nativescript-community/sentry
ns run android
```
No errors will be reported by Sentry

### Is there any code involved? 
`app/app.js`
 ```
import * as Sentry from '@nativescript-community/sentry';
import Vue from 'nativescript-vue'

import Home from './components/Home'

Sentry.init({
  dsn: "---your-DSN---"
});

Vue.config.silent = false;

try {
  foo.bar();
  // potentially crashy code
} catch (error) {
  console.log('error', error);
  Sentry.captureException(error);
}

new Vue({
  render: (h) => h('frame', [h(Home)]),
}).$start()
```

**Comment by farfromrefug** [comment]
Cant help you much with such a report. You need to debug and see is something is sent from your phone, but also if something is received by your server

**Comment by sebj54** [comment]
Hi @farfromrefug, I'm currently having this issue too. Nothing seems to be sent from my phone when I'm using `captureException`.

I would like to help, but how can I be sure nothing is sent? From the Netwotk tab in the devtools? Also, `captureException` returns an event id as expected.

**Comment by farfromrefug** [comment]
@sebj54 is your sentry setup fully working ? Can you confirm it receives event?
What is your sentry server version ?
You need to share  all your setup

**Comment by sebj54** [comment]
Thanks for your quick answer :)

I have a local Sentry server (v 21.7.0) but I also tested with the Sentry Saas (https://sentry.io). 

I made a specific app to debug this problem easily: https://github.com/sebj54/ns-debug-sentry. Maybe we can reuse it for the demo app once the problem is solved? The actual is made with {N} 7 if I'm right.

I also invited you the join the Sentry project so you can see there is no issue reported.

**Comment by farfromrefug** [comment]
@sebj54 yes it is possible for the errors to be sent on the second run. The reason is that the app has crashed => stopped and cant restart itself. It is the normal behavior. Sending messages should be instant though. Can we close this? Btw if you think your demo demonstrate well how to use this plugin could you create a PR to replace the default one ?

**Comment by sebj54** [comment]
I'm not sure we can close because when we call `captureException`, it does not crash and it will probably never crash. So it is not very convenient to wait maybe 24 hours before getting any report.

I'm not sure but I think the report also never gets reported if the app restarts without crashing (I will make some tests today or tomorrow and let you know).

**Comment by sebj54** [comment]
For the last test, I waited 30 minutes after hitting the button "capture exception" in my test app. After this, I hit the "throw exception" button, the app crashed and only after I relaunched the app, both exception reports were sent.

The server should be strong enough because I'm using sentry.io (official Saas for Sentry) now to eliminate the server from the problem.

**Comment by farfromrefug** [comment]
@sebj54 ok good for the server. You need to use something lik httptoolkit to ensure the message is not sent right away. For me it is. After that you can try and bump the native lib version and check the issues on their repo. The sending part is done natively i have no control over it

**Comment by sebj54** [comment]
That's what I thought. I am going to try httptoolkit and debug this deeper tomorrow, I'll let you know.

**Comment by sebj54** [comment]
Hi @farfromrefug, I just tested my demo app with httptoolkit and I can confirm nothing is sent right after `captureException` call. I have to kill the app and relaunch it (the app does not need to crash).

I tested with a local version of your plugin and updated all Sentry packages to 6.12.0. It still send reports on the next app launch.

I don't know why it is different in your case. Did you test it from my app? Maybe I missed a configuration?

I also wanted to test it with `debug: true`, but it crashes (`global.console`  is undefined in `consoleSandbox`):

```
JS: TypeError: Cannot read property 'log' of undefined
JS:     at file: app/webpack:/@nativescript/nativescript-sentry/node_modules/@sentry/utils/esm/logger.js:31:0
JS:     at consoleSandbox (file: app/webpack:/@nativescript/nativescript-sentry/node_modules/@sentry/utils/esm/misc.js:101:0)
JS:     at Logger.log (file: app/webpack:/@nativescript/nativescript-sentry/node_modules/@sentry/utils/esm/logger.js:30:22)
JS:     at setupIntegration (file: app/webpack:/@nativescript/nativescript-sentry/node_modules/@sentry/core/esm/integration.js:46:4)
JS:     at file: app/webpack:/@nativescript/nativescript-sentry/node_modules/@sentry/core/esm/integration.js:58:0
JS:     at Array.forEach (<anonymous>)
JS:     at setupIntegrations (file: app/webpack:/@nativescript/nativescript-sentry/node_modules/@sentry/core/esm/integration.js:56:0)
JS:     at NativescriptClient.BaseClient.setupIntegrations (file: app/webpack:/@nativescript/nativescript-sentry/node_modules/@sentry/core/esm/baseclient.js:154:50)
JS:     at Hub.bindClient (file: app/webpack:/@nativescript/nativescript-sentry/node_modules/@sentry/hub/esm/hub.js:57:0)
JS:     at initAndBind (file: app/webpack:/@nativescript/nativescript-sentry/node_modules/@sentry/core/esm/sdk.js:18:0)
```

**Comment by erodriguezh** [comment]
I can also reproduce the `debug: true` error on

* Android 11 (emulator and device)
* iOS 14.5 (emulator)

Please, provide the following version numbers that your issue occurs with:

  CLI: 8.0.2
  Cross-platform modules: 8.0.0
  Runtime(s): Android 8.0.0
  
```json
  "dependencies": {
    "@angular/animations": "~12.0.0",
    "@angular/common": "~12.0.0",
    "@angular/compiler": "~12.0.0",
    "@angular/core": "~12.0.0",
    "@angular/forms": "~12.0.0",
    "@angular/platform-browser": "~12.0.0",
    "@angular/platform-browser-dynamic": "~12.0.0",
    "@angular/router": "~12.0.0",
    "@auth0/angular-jwt": "^5.0.2",
    "@bradmartin/nativescript-urlhandler": "^2.0.1",
    "@contentful/rich-text-html-renderer": "^14.1.2",
    "@nativescript-community/extendedinfo": "^1.2.1",
    "@nativescript-community/sentry": "^3.0.0",
    "@nativescript-community/ui-lottie": "^1.0.0",
    "@nativescript-community/ui-pulltorefresh": "^2.4.8",
    "@nativescript/angular": "~12.0.0",
    "@nativescript/core": "~8.0.0",
    "@nativescript/firebase": "^11.1.3",
    "@nativescript/iqkeyboardmanager": "^2.0.0",
    "@nativescript/theme": "~3.0.1",
    "@ngx-translate/core": "~13.0.0",
    "@triniwiz/nativescript-toasty": "^4.1.3",
    "nativescript-carousel": "^7.0.1",
    "nativescript-inappbrowser": "^3.0.2",
    "nativescript-oauth2": "^3.0.6",
    "reflect-metadata": "~0.1.13",
    "rxjs": "^6.6.7",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular/compiler-cli": "~12.0.0",
    "@nativescript/android": "8.0.0",
    "@nativescript/ios": "8.0.0",
    "@nativescript/types": "~8.0.0",
    "@nativescript/webpack": "beta",
    "@ngtools/webpack": "~12.0.0",
    "typescript": "~4.2.0"
  }
```

**Comment by farfromrefug** [comment]
Those issues are mostly always related to your setup/config.
It is almost impossible to help you without you making further investigations about your setup and ensuring everything is set up correctly(and updated)

**Comment by erodriguezh** [comment]
Hi @farfromrefug,

Thank you for answering so quickly. 
Could you please give me some pointers as to what should I specifically check?
I do not mind doing the field work so long I can get some help.

**Comment by farfromrefug** [comment]
@erodriguezh you need to ensure your sentry instance can receive messages (you should be able to trigger them from another sdk or even by command line).
Then ensure all you config, tokens, etc... are set up correctly on N side.
You can then check if something is "getting" out of the android sdk using httptoolkit. 
Also if you testing "crash" report remember to re open the app after the crash. Most of the times crash reports are written on disk and sent on next app launch.
THose are the kind of steps you can take to check it all

**Comment by sebj54** [comment]
> Also if you testing "crash" report remember to re open the app after the crash. Most of the times crash reports are written on disk and sent on next app launch.

@erodriguezh I confirm that you should try to exit the app and relaunch it. Every report is sent on app launch in my case.

@farfromrefug If this is the expected behavior, then my issue is no longer relevant and should be closed. But the fact that this is happening even with manual reports (like `captureException`) makes me think it does not work totally as expected.

**Comment by farfromrefug** [comment]
@sebj54 yes it is the expected behavior. I actually using a `flush` method before to try and send reports right away. But the `flush` method is not working anymore... I know they want to bring it back but we dont know when

**Comment by sebj54** [comment]
Hi @farfromrefug, I'm actually trying to fix this "problem" once for all. My goal here is to add an option to instantly send exceptions or manual reports.

Do you have any links to issues (or anything that could help help) about this non-working flush method? In my app, the `flush` method always resolves but events are still not sent.

**Comment by farfromrefug** [comment]
@sebj54 it is the native SDK flush which did not work (not sure if it changed). You need to search in the sentry native sdks

**Comment by sebj54** [comment]
Ok thanks for your quick answer, I'll check the native repos so.

But actually I'm trying to instantly report JS exceptions reported with `captureException` so there might be a problem also with JS SDK.

**Comment by sebj54** [comment]
You're right, that's what I noticed too.

If anyone thinks it should be reopened, feel free to ask.
